src/main/java/com/openshift/internal/restclient/model/DeploymentConfig.java

```diff
@@ -164,4 +188,79 @@ public IDeploymentTrigger addTrigger(String type) {
 	public String getDeploymentStrategyType() {
 		return asString(DEPLOYMENTCONFIG_STRATEGY);
 	}
+
+	@Override
+	public boolean haveTriggersFired() {
+		ModelNode causes = get(DEPLOYMENTCONFIG_CAUSES);
+		if (causes == null || causes.getType() != ModelType.LIST)
+			return false;
+		return causes.asList().size() > 0;
+	}
+
+	@Override
+	public boolean didImageTrigger(String imageTag) {
```

Comment:
my initial interpretation of the restclient data model was that they were generally objects at the granularity of items retrievable via `oc get` (though IBuildRequest was an exception which made sense to me).  Deployment causes would not qualify under that rationale.  So my initial reaction was that it would be overkill, but I for sure would not qualify my opinion here as resolute.